stdlib/public/core/RangeReplaceableCollectionType.swift

```diff
@@ -446,10 +446,10 @@ public func +<
     S : SequenceType
     where S.Generator.Element == C.Generator.Element
 >(lhs: C, rhs: S) -> C {
-  var lhs = lhs
-  // FIXME: what if lhs is a reference type?  This will mutate it.
-  lhs.appendContentsOf(rhs)
-  return lhs
+  var result = lhs is AnyObject ? C(lhs) : lhs
```

Comment:
Yeah in that sense, true, an arbitrary type conforming to `RangeReplaceableCollectionType` doesn't necessarily perform that optimization. Though, I'd still argue calling `C(lhs)` is the way to go:
- All stdlib collection types do perform this optimization, and many custom collection types are built on top of them so 'inherit' the optimization.
- The behavior does not become semantically *wrong* if a custom type doesn't implement it.
- That initializer performs a semantic copy, which is the behavior we need in the general case, because semantically, we're returning a new collection.
Or do you have a better alternative in mind? Because the current behavior is just plain incorrect for reference types, which I'd argue is much worse.